src/patterns/filtering/multiple-categories.stories.ts

```diff
@@ -0,0 +1,226 @@
+        </div>
+    </div>
+    `,
+    styles: [`
+        .header {
+            margin-bottom: 80px;
+        }
+
+        .reset-button {
+            margin-bottom: 20px;
+        }
+
+        .data-table {
+            width: 100%;
+        }
+    `
+    ]
+})
+class SampleMultiCategories {
+    model = new TableModel();
+    checkBoxFilters = new Set();
+    radioFilter = null;
+
+    dataset = [
+        { name: "Apple", type: ["Fruit"], color: "Red" },
+        { name: "Grape", type: ["Fruit"], color: "Purple" },
+        { name: "Eggplant", type: ["Fruit"], color: "Purple" },
+        { name: "FruitVegMeat", type: ["Fruit", "Vegetable", "Meat"], color: "White" },
+        { name: "Lettuce", type: ["Vegetable"], color: "Green" },
+        { name: "Daikon Radish", type: ["Vegetable"], color: "White" },
+        { name: "Beef", type: ["Meat"], color: "Red" }
+    ];
+
+    radios = [
+        { color: "Red", checked: false },
+        { color: "Purple", checked: false },
+        { color: "Green", checked: false },
+        { color: "White", checked: false }
+    ]
+
+    checkboxList = [
+        { value: "Fruit", checked: false},
+        { value: "Vegetable", checked: false},
+        { value: "Meat", checked: false}
+    ]
+
+    onCheckboxChange(event) {
+        if (event.checked) {
+            this.checkBoxFilters.add(event.source.value);
+        } else {
+            this.checkBoxFilters.delete(event.source.value);
+        }
+        this.applyFilters();
+    }
+
+    onRadioChange(event) {
+        this.radioFilter = event.value;
+        this.applyFilters();
+    }
+
+    resetFilters() {
+        this.resetCheckboxList();
+        this.resetRadios();
+    }
+
+    resetCheckboxList() {
+        this.checkBoxFilters = new Set();
+        this.checkboxList = this.checkboxList.map(obj => {
+            return { value: obj.value, checked: false }
+        });
+        this.applyFilters()
+    }
+
+    resetRadios() {
+        this.radioFilter = null;
+        this.radios = this.radios.map(obj => {
+            return { color: obj.color, checked: false }
+        });
+        this.applyFilters();
+    }
+
+    applyFilters() {
+        this.model.data = 
+            this.dataset
+                .filter(data =>
+                    (this.applyMultiFilters(data.type, this.checkBoxFilters) || !this.checkBoxFilters.size) &&
+                    (data.color === this.radioFilter || !this.radioFilter))
+                .map(filteredData =>                     
+                    [
+                        new TableItem({ data: filteredData.name }),
+                        new TableItem({ data: "Lorem ipsum dolor sit amet, consectetur adipiscing elit." })
+                    ]);
+    }
+
+    // Apply filters of a multi selection, returns true if the given items pass the filter and false if not.
+    applyMultiFilters(types: Array<String>, filters): Boolean {
+        let shouldInclude = true;
+        filters.forEach(filter => {
```

Comment:
Any reason why this whole thing wouldn't just use `some`?

Reply:
It needs to satisfy all filters, so I will use `every`